### PR TITLE
Add: Test beahviour where json has null keys

### PR DESCRIPTION
**Pull request by nickanderson**

### CODE REVIEW

**Comment by tzz** [comment]
This is good, but you can simplify it nicely with

```
bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
```

see for example `tests/acceptance/00_basics/01_compiler/string_contexts.cf`

File: tests/acceptance/01_vars/04_containers/test_json_with_and_without_nulls.cf [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    "test_soft_fail"
+      string => "any",
+      meta => { "redmine7821" };
+
+  methods:
+      "ec2"
+	comment => "We activate the bundle so that we can grab its state";
+  vars:
+      "state" data => bundlestate(ec2);
```
Comment (tzz) [inline tests/acceptance/01_vars/04_containers/test_json_with_and_without_nulls.cf]: You don't need this for the `dcs_check_state()` bundle.

File: tests/acceptance/01_vars/04_containers/test_json_with_and_without_nulls.cf [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  files:
+    "$(G.testfile)"
```
Comment (tzz) [inline tests/acceptance/01_vars/04_containers/test_json_with_and_without_nulls.cf]: You can skip this too, it's done automatically by `dcs_check_state()`

**Comment by nickanderson** [comment]
@tzz I updated the test after your comments, thanks.

**Comment by tzz** [comment]
Cool, thanks.
